src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -126,8 +124,11 @@ private void OnFailedRequest(string url, Exception e)
         {
             FailedRequest?.Invoke(url, e);
         }
+        
+        private HttpClient GetCachedOrCreateHttpClient() => 
+            GlobalHttpClient.GetOrAdd(Certificate?.Thumbprint ?? string.Empty, new Lazy<HttpClient>(CreateClient)).Value;
```

Comment:
You are creating this globally, so another document store with compression false will use compression